tests/unittest/_torch/pyexecutor/test_profile_endpoints.py

```
@@ -0,0 +1,212 @@
+
+
+def test_stop_profile_cancels_pending_start_before_firing():
+    """If ``start_profile()`` was scheduled but the engine has not yet
+    reached ``start_iter`` (idle server), ``stop_profile()`` must remove
+    the pending start so profiling does not silently begin later."""
+    executor = _bare_executor()
+    executor.iter_counter = 3
+
+    # Simulate the full ``start_profile`` path for a bare executor: the
+    # broadcast apply is what populates ``profile_start_iters``.
+    executor.start_profile(output_dir="/tmp/unused", num_steps=50)
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/unused"

(S108)